### PR TITLE
Fix: incorrect call to .tail on a possibly empty list

### PR DESCRIPTION
**Pull request by shtukas**
This change corrects a runtime bug appearing when a user doesn't have at least one curation or verification step.

### CODE REVIEW

File: common/src/main/scala/com/gu/recipeasy/models/Statistics.scala [hunk 1]
```diff
@@ -71,16 +71,24 @@ object UsersSpeedsMeasurements {
       val userEvents = userEventsAll.filter(event => event.user_email == email)
 
       val curationEvents = userEvents.filter(event => (event.operation_type == UserEventAccessRecipeCurationPage.name || event.operation_type == UserEventCuration.name))
-      val curationTimings: List[Double] = curationEvents.zip(curationEvents.tail)
-        .filter(events => pairOfEventsIsNormalised(events, UserEventAccessRecipeCurationPage.name, UserEventCuration.name))
-        .map(events => secondsBetweenEvents(events._1, events._2))
-        .filter(timing => timing <= 1200)
+      val curationTimings: List[Double] = if (!curationEvents.isEmpty) {
```
Comment (LATaylor-guardian) [inline common/src/main/scala/com/gu/recipeasy/models/Statistics.scala]: `if (curationEvents.nonEmpty) {` is preferred.

File: common/src/main/scala/com/gu/recipeasy/models/Statistics.scala [hunk 2]
```diff
@@ -71,16 +71,24 @@ object UsersSpeedsMeasurements {
...
+      } else {
+        List()
+      }
 
       val verificationEvents = userEvents.filter(event => (event.operation_type == UserEventAccessRecipeVerificationPage.name || event.operation_type == UserEventVerification.name))
-      val verificationTimings: List[Double] = verificationEvents.zip(verificationEvents.tail)
-        .filter(events => pairOfEventsIsNormalised(events, UserEventAccessRecipeVerificationPage.name, UserEventVerification.name))
-        .map(events => secondsBetweenEvents(events._1, events._2))
-        .filter(timing => timing <= 1200)
+      val verificationTimings: List[Double] = if (!verificationEvents.isEmpty) {
```
Comment (LATaylor-guardian) [inline common/src/main/scala/com/gu/recipeasy/models/Statistics.scala]: As above.

File: common/src/main/scala/com/gu/recipeasy/models/Statistics.scala [hunk 3]
```diff
@@ -71,16 +71,24 @@ object UsersSpeedsMeasurements {
...
-        .filter(events => pairOfEventsIsNormalised(events, UserEventAccessRecipeCurationPage.name, UserEventCuration.name))
-        .map(events => secondsBetweenEvents(events._1, events._2))
-        .filter(timing => timing <= 1200)
+      val curationTimings: List[Double] = if (!curationEvents.isEmpty) {
+        curationEvents.zip(curationEvents.tail)
+          .filter(events => pairOfEventsIsNormalised(events, UserEventAccessRecipeCurationPage.name, UserEventCuration.name))
+          .map(events => secondsBetweenEvents(events._1, events._2))
+          .filter(timing => timing <= 1200)
+      } else {
+        List()
```
Comment (tomrf1) [inline common/src/main/scala/com/gu/recipeasy/models/Statistics.scala]: I think `Nil` is more idiomatic

File: common/src/main/scala/com/gu/recipeasy/models/Statistics.scala [hunk 4]
```diff
@@ -71,16 +71,24 @@ object UsersSpeedsMeasurements {
...
-        .filter(events => pairOfEventsIsNormalised(events, UserEventAccessRecipeVerificationPage.name, UserEventVerification.name))
-        .map(events => secondsBetweenEvents(events._1, events._2))
-        .filter(timing => timing <= 1200)
+      val verificationTimings: List[Double] = if (!verificationEvents.isEmpty) {
+        verificationEvents.zip(verificationEvents.tail)
+          .filter(events => pairOfEventsIsNormalised(events, UserEventAccessRecipeVerificationPage.name, UserEventVerification.name))
+          .map(events => secondsBetweenEvents(events._1, events._2))
+          .filter(timing => timing <= 1200)
+      } else {
+        List()
```
Comment (LATaylor-guardian) [inline common/src/main/scala/com/gu/recipeasy/models/Statistics.scala]: Nil is more idiomatic and should be preferred.

File: common/src/main/scala/com/gu/recipeasy/models/Statistics.scala (hunk 3 above)
Comment (LATaylor-guardian) [inline common/src/main/scala/com/gu/recipeasy/models/Statistics.scala]: Nil is more idiomatic and should be preferred.

**Comment by LATaylor-guardian** [comment]
Other than comments 👍

**Comment by shtukas** [comment]
All fixed. Merging now.

**Pull request merged**